experiments/modern/src/App.js

```diff
@@ -213,23 +219,38 @@ function XmlNode({ node }) {
     }
     return null;
   }
-  return <Component {...attributes}>{children}</Component>;
+  return <Component node={node} {...attributes}>{children}</Component>;
 }
 
 function App() {
   const [data, setData] = React.useState(null);
   React.useEffect(() => {
-    getSkin().then(setData);
+    getSkin().then(async ({ root, registry }) => {
+      // Execute scripts
+      await Utils.asyncTreeFlatMap(root, async node => {
+        switch (node.xmlNode.name) {
+          case "script": {
+            const scriptGroup = Utils.findParentNodeOfType(node, ["Group", "WinampAbstractionLayer"]);
```

Comment:
I would love to find an API where we pass the actual class rather than the name. That would help prevent typos, since the import of the class would fail.

Reply:
I agree that would be better. Is there a "sane" way to do it without importing every Maki object into each place we want to do that though?